test/with_dummyserver/test_socketlevel.py

```diff
@@ -1133,6 +1133,34 @@ def echo_socket_handler(listener: socket.socket) -> None:
                     retries=False,
                 )
 
+    def test_alpn_protocol_in_first_request_packet(self) -> None:
+        self.buf = b""
+
+        def echo_socket_handler(listener: socket.socket) -> None:
+            sock = listener.accept()[0]
+            self.buf = b""
+            while not self.buf.endswith(b"\r\n\r\n"):
+                self.buf += sock.recv(65536)
+
+            sock.send(
+                (
+                    "HTTP/1.1 200 OK\r\n"
+                    "Content-Type: text/plain\r\n"
+                    "Content-Length: %d\r\n"
+                    "\r\n"
+                    "%s" % (len(self.buf), self.buf.decode("utf-8"))
+                ).encode("utf-8")
+            )
+            sock.close()
+
+        self._start_server(echo_socket_handler)
+        base_url = f"http://{self.host}:{self.port}"
```

Comment:
This is using HTTPS, tunneling and ALPN are TLS features so requires HTTPS. It's expected for the handshake to fail, we want to see what ALPN values are being offered and not complete the handshake.